packages/@aws-cdk/aws-eks/lib/helm-chart.ts

```diff
@@ -13,9 +15,9 @@ import { Construct as CoreConstruct } from '@aws-cdk/core';
 
 export interface HelmChartOptions {
   /**
-   * The name of the chart.
+   * The name of the chart or an Asset.
    */
-  readonly chart: string;
+  readonly chart: string | Asset;
```

Comment:
We don't use union types since this translate poorly in other languages (namely Java and C#). 

Instead, you'll have to create a different property called `chartAsset` (or something like that